File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
import java.util.*;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        int[] ans = new int[photo.length];
        Map<String, Integer> map = new HashMap<>();
        
        for (int i = 0; i < name.length; i++) {
            map.put(name[i], yearning[i]);
        }
        
        for (int i = 0; i < photo.length; i++) {
            for (int j = 0; j < photo[i].length; j++) {
                ans[i] += map.getOrDefault(photo[i][j], 0);
            }
        }
        return ans;
    }
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    const map = name.reduce((acc, cur, index) => {
        acc[cur] = yearning[index];
        return acc;
    }, {});

    return photo.map(e => {
        return e.reduce((acc, cur) => {
            acc += map[cur] || 0;
            return acc;
        }, 0)
    });
}
